### PR TITLE
Provider db config (task #2048)

### PR DESCRIPTION
**Pull request by eliamariutti**
Add new Config Engine that read configuration from database

### CODE REVIEW

File: src/Settings/DBConfig.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?php
+namespace App\Settings;
+
+use Cake\Core\Configure\ConfigEngineInterface;
+use Cake\ORM\TableRegistry;
+use Cake\Utility\Hash;
+
+class DBConfig implements ConfigEngineInterface
```
Comment (softius) [inline src/Settings/DBConfig.php]: @eliamariutti Should be `DbConfig`

File: src/Settings/DBConfig.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    /**
+     * @param string $key Table name
+     * @param array $data Data to dump.
+     * @return bool Success
+     */
+    public function dump($key, array $data)
+    {
+        return false;
```
Comment (softius) [inline src/Settings/DBConfig.php]: @eliamariutti please consider throwing an exception i.e. `throw new Exception('Not implemented');`

File: config/bootstrap.php [hunk 3]
```diff
@@ -308,3 +321,5 @@
  * Register custom database type(s)
  */
 Type::map('base64', 'App\Database\Type\EncodedFileType');
+
+debug(Configure::read());
```
Comment (softius) [inline config/bootstrap.php]: @eliamariutti please remove any debugging statements

File: config/bootstrap.php [hunk 4]
```diff
@@ -167,6 +168,18 @@
...
 Email::config(Configure::read('Email'));
 
+/**
+ * After the connection manager is set up it's possible
+ * to load the dbconfig engine and merge it with the other
+ * configuration.
+ */
+try {
+    Configure::config('dbconfig', new DBConfig());
+    Configure::load('Settings1', 'dbconfig', true);
```
Comment (softius) [inline config/bootstrap.php]: @eliamariutti Why `Settings1`? was expecting this to be `Settings`

Comment (georgeconstantinou) [inline config/bootstrap.php]: @eliamariutti for merging configurations [this](https://github.com/QoboLtd/cakephp-csv-migrations/blob/master/config/bootstrap.php#L9-L23) might help.

File: config/bootstrap.php [hunk 5]
```diff
@@ -53,6 +54,7 @@
 use Cake\Log\Log;
 use Cake\Network\Email\Email;
 use Cake\Network\Request;
+use Cake\ORM\TableRegistry;
```
Comment (georgeconstantinou) [inline config/bootstrap.php]: @eliamariutti this class is not used.

**Pull request merged**